server/src/api/UserInterestRoutes.ts

```diff
@@ -0,0 +1,134 @@
+import express from "express";
+import { authenticateToken } from "../middlewares";
+import { prisma } from "../lib/prisma";
+import { INTERNAL_ERROR_CODE, INVALID_REQUEST_CODE, OK_CODE } from "../lib/StatusCodes";
+import { Category, UserInterest } from "@prisma/client";
+
+const router = express.Router();
+
+type InterestsResponse = {
+  faculties: Category[],
+  ethnicities: Category[],
+  religions: Category[],
+  programs: Category[],
+  hobbies: Category[]
+}
+
+type UserInterestsRequest = {
+  faculty: number,
+  ethnicity: number,
+  religion: number,
+  program: number,
+  hobbies: number[]
+}
+
+type UserInterestsResponse = {
+  interests: UserInterest[]
+}
+
+type UserInterestWithoutId = {
+  user_id: number,
+  category_id: number
+}
+
+router.get<void, InterestsResponse>("/all", authenticateToken, async (req, res) => {
+  try {
+    const promises = [
+      prisma.category.findMany({
+        where: {type: 'faculty'}
+      }),
+      prisma.category.findMany({
+        where: {type: 'ethnicity'}
+      }),
+      prisma.category.findMany({
+        where: {type: 'religion'}
+      }),
+      prisma.category.findMany({
+        where: {type: 'program'}
+      }),
+      prisma.category.findMany({
+        where: {type: 'hobby'}
+      })
+    ];
+
+    const [faculties, ethnicities, religions, programs, hobbies] = await Promise.all(promises);
+
+    res.status(OK_CODE).json({faculties, ethnicities, religions, programs, hobbies});
+  } catch(error) {
+    res.sendStatus(INTERNAL_ERROR_CODE);
+  }
+});
+
+router.get<void, UserInterestsResponse>("/", authenticateToken, async (req, res) => {
+  try {
+    const userId = req.body.user.id;
+
+    const interests: UserInterest[] = await prisma.userInterest.findMany({
+      where: {
+        user_id: userId
+      },
+      include: {
+        category: true
+      }
+    })
+
+    res.status(OK_CODE).json({interests});
```

Comment:
- remove the {} outside of interests
- you should format this to flatten, rn its a weird format that's hard to work with in the UI, make sure the response type looks like this:
{user_id: Int, category_id: Int, category_name: String, type: String}[]

Reply:
acc it could just be {category_id: Int, category_name: String, type: String}[], they'll already have the user id in the UI

Reply:
I have done this instead:

`{
  category_id: 7,
  category: {
    type: 'faculty',
    name: 'Arts'
  }
}`

Reply:
this is not just as good, we already have a Category data class in the ClubDetails.kt side that's being used. we should align with that to not have unnessecary new data classes that do the same thing.

<img width="203" alt="Screenshot 2024-03-11 at 7 25 27 PM" src="https://github.com/Kggupta/cs446-project/assets/33162944/91535a92-6b19-42d1-9910-de7ce5c92b20">

Reply:
@Kggupta  ok, noted, will do.

Reply:
resolved.